src/sage/rings/number_field/order.py

```diff
@@ -97,25 +97,79 @@
 from sage.structure.sequence import Sequence
 
 
-def quadratic_order_class_number(disc):
+def quadratic_order_class_number(disc, proof=None):
```

Comment:
```suggestion
def quadratic_order_class_number(disc, *, proof=None):
```
(Suggestion.) This is to make sure callers have to explicitly pass the `proof` argument as a keyword argument, rather than a nondescript positional argument.